lib/Manager.php

```diff
@@ -247,10 +247,10 @@ public function getAnnouncement($id, $parseStrings = true, $ignorePermissions =
 	 */
 	public function getAnnouncements($limit = 15, $offset = 0, $parseStrings = true) {
 		$query = $this->connection->getQueryBuilder();
-		$query->select('a.*')
+		$query->select('a.announcement_id', 'a.announcement_time', 'a.announcement_user', 'a.announcement_subject', 'a.announcement_message', 'a.allow_comments')
 			->from('announcements', 'a')
 			->orderBy('a.announcement_time', 'DESC')
-			->groupBy('a.announcement_id')
+			->groupBy('a.announcement_id', 'a.announcement_time', 'a.announcement_user', 'a.announcement_subject', 'a.announcement_message', 'a.allow_comments')
```

Comment:
From how I read the docs, adding `'a.announcement_time'` is enough.

It only needs all fields which are used in orderBy (`'a.announcement_time'`)  and having (which we dont have)

Reply:
The doc says:
> Reject queries for which the **select list**, HAVING condition, or ORDER BY list ...

So the fields from the `select` method must be specified within the `groupBy` method. If I perform the query with only the `a.announcement_time` as additional field in the `GROUP BY` list I get this error:
```
MariaDB [nextcloud]> SELECT `a`.* FROM `oc_announcements` `a` LEFT JOIN `oc_announcements_groups` `ag` ON `a`.`announcement_id` = `ag`.`announcement_id` WHERE `ag`.`gid` IN ('***', 'everyone') GROUP BY `a`.`announcement_id`, `a`.`announcement_time` ORDER BY `a`.`announcement_time` DESC LIMIT 5\G                                                                                                          
ERROR 1055 (42000): 'nextcloud.a.announcement_user' isn't in GROUP BY
```

Reply:
Ah, misread that. I will have a look again

Reply:
any news?

Reply:
Sorry for the late response, I was unavailable the last month for medical reasons and had important tasks on my desk which needed to be dealt with first.